### DISCUSSION TITLE
Add Tailwind to a Solidus store

### DISCUSSION THREAD

**Discussion by gabrielguerin** [solidusio/solidus#4731] [Troubleshooting] [answered]
Hello everyone!

I have been trying to **add Tailwind to a Solidus store** with no luck. Adding Tailwind the usual way you would do on any Rails app does not work, as Tailwind components do not display properly. Unfortunately, I haven't found any guide on how to achieve this.

Basically, I would like to apply changes to [Solidus' Starter Frontend](https://github.com/solidusio/solidus_starter_frontend) using Tailwind. 
Adding Tailwind's CDN to `storefront.html.erb` seems to work however (with the text being very small).

**Steps**

1. I have followed every steps on the [Getting Started](https://guides.solidus.io/getting-started/installing-solidus) page from Solidus' official documentation, but with `rails new amazing_store --css tailwind --skip-javascript` instead of just `rails new amazing_store --skip-javascript`.
2. Then, when prompted, I chose to install the [solidus_starter_frontend](https://github.com/solidusio/solidus_starter_frontend) storefront.

I have checked [this repository](https://github.com/gsmendoza/solidus_store/tree/rails-7-0-4--solidus-3-2-2--4685-tailwind) (found on [this GitHub issue](https://github.com/solidusio/solidus/issues/4685)) and compared it with my own code and I cannot find any difference with mine.

Do you have any idea on how to achieve this?

**Comment by kennyadsl** [comment, marked as answer]
Hello!

I have a Solidus store running with Tailwind. Here's the step I followed to make it work:


```
rails new amazing_store # avoid the --skip-javascript flag here
cd amazing_store
bundle add 'solidus'
bin/rails generate solidus:install # devise: Y, payment: none, frontend: starter
bundle add tailwindcss-rails
rails tailwindcss:install
```


In `layout/storefront.html.erb`:

```diff
-    <%= stylesheet_link_tag 'solidus_starter_frontend', media: 'screen' %>
+    <%= stylesheet_link_tag "tailwind", "inter-font", "data-turbo-track": "reload" %>
```

Now start rails with:

```sh
bin/dev
```

At this point you should be able to add tailwind classes to your HTML; all changes to related files will trigger a recompile, and a page reload in the browser automatically.

**Reply by gabrielguerin** [reply]
Thanks @kennyadsl, your step-by-step works perfectly!
